sdk/core/azure-core/CMakeLists.txt

```diff
@@ -117,6 +117,7 @@ set(
     src/logger.cpp
     src/operation_status.cpp
     src/strings.cpp
+    src/internal/keyvault_exception.cpp
```

Comment:
Don't use this as the file or type name, and instead add this to the `exception.hpp`